src/main/java/mate/academy/dao/impl/UserDaoImpl.java

```diff
@@ -19,7 +19,7 @@ public UserDaoImpl(SessionFactory sessionFactory) {
     public Optional<User> findByEmail(String email) {
         try (Session session = sessionFactory.openSession()) {
             return session.createQuery(
-                    "from User u join fetch u.roles where u.email = :email", User.class)
+                    "from User u left join fetch u.roles where u.email = :email", User.class)
```

Comment:
what for?

Reply:
REF to screens attached
1) Method fails if I try to save user with Set of Roles
2) findByEmail return Empty if I save without roles but leave just "join fetch" 
Any suggestions, please? 
![Screenshot_2](https://user-images.githubusercontent.com/74661731/165048533-34b0840a-b536-418e-bef8-918244be0185.jpg)
![Screenshot_1](https://user-images.githubusercontent.com/74661731/165048540-898bd421-bfa2-42d6-99c9-5b3cf31511fb.jpg)

Reply:
You can't save user with role, that you doesn't have in DB, you should find this role in DB first and then set it in user